### ISSUE TITLE
Docker not installing on Amazon Linux 2 with latest AMIs 

### ISSUE THREAD

**Issue by adybuxton** [chef-cookbooks/docker#1039]
### Cookbook version
4.6.8

### Chef-client version
13.6.4

### Platform Details
Amazon Linux 2 (Latest AMIs)

### Scenario:
Install docker

### Steps to Reproduce:
Install docker

### Expected Result:
Docker is installed

### Actual Result:
Chef::Exceptions::Package: No candidate version available for docker  

This is due to Amazon Linux moving docker packages away from the standard yum library to the Amazon Linux Extras library

**Comment by ChrisG1971** [comment]
Is this what is causing the install to fail?  I am using 4.8.0 and it still fails passing and doesn't even reference the version I'm passing as the package.  

 Error executing action `create` on resource 'docker_installation_package[default]'
    ================================================================================

    Chef::Exceptions::Package
    -------------------------
    yum_package[docker-ce] (/var/cache/chef/cookbooks/docker/libraries/docker_installation_package.rb line 42) had an error: Chef::Exceptions::Package: Version ["18.06.1"] of ["docker-ce"] not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)

    Resource Declaration:
    ---------------------
    # In /var/cache/chef/cookbooks/wdpr_docker/recipes/docker.rb

     22: docker_installation_package 'default' do
     23:   package_name 'docker-ce'
     24:   package_version node['wdpr_docker']['version']
     25:   action :create
     26:   package_options %q|--force-yes -o Dpkg::Options::='--force-confold' -o Dpkg::Options::='--force-all'|
     27: end
     28:

default['wdpr_docker']['version'] = '18.09.0-3.el7'

**Comment by tas50** [comment]
The docker packages aren't actually compatible with Amazon Linux 2. They don't have an Amazon specific repo and even if you setup the CentOS 7 repo it attempts to install packages using the amazon naming and then fails. There's no way around this until Docker officially supports Amazon Linux 2. Until then you'll want to use one of the other installation methods or switch to another platform.

**Comment by doertedev** [comment]
As of writing this, `amazon-linux-extras install docker` will install docker. I execute it via the execute resource. Ugly but works.

**Comment by megamorf** [comment]
FYI: 18.09 is not yet available on AL2:
```
$ sudo amazon-linux-extras list | grep -C 1 docker
 19  gimp                     available    [ =2.8.22 ]
 20  docker=latest            enabled      \
        [ =17.12.1  =18.03.1  =18.06.1 ]
```
